HV: refine copy_from_vm/copy_to_vm

YadongQi opened the pull request:
If the target is an array, then only the first element
will be copied.
So replace structure assignment with memcpy_s().

Signed-off-by: Qi Yadong <yadong.qi@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

kwang13 commented on hypervisor/arch/x86/trusty.c:
```diff
@@ -360,6 +360,10 @@ static bool setup_trusty_info(struct vcpu *vcpu,
 
+	/* Copy virtual RPMB Key to trusty's memory */
+	memcpy_s(&mem->first_page.key_info.rpmb_key[0][0], RPMB_KEY_LEN,
+			&vcpu->vm->vRPMB_KEY[0], RPMB_KEY_LEN);
```
set 0 to other PRMB keys

kwang13 commented on hypervisor/arch/x86/trusty.c:
```diff
@@ -456,3 +460,29 @@ bool initialize_trusty(struct vcpu *vcpu, uint64_t param)
+	 */
+	if (is_vm0(vm)) {
+		memcpy_s(hva, RPMB_KEY_LEN,
+			&g_key_info.rpmb_key[0][0], RPMB_KEY_LEN);
```
store the real rpmb in SOS's VM->RPMB_KEY[] (use the name "RPMB_KEY" instead of "vRPMB_KEY")

YadongQi committed:
HV: refine copy_from_vm/copy_to_vm

If the target is an array, then only the first element
will be copied.
So replace structure assignment with memcpy_s().

Signed-off-by: Qi Yadong <yadong.qi@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

YadongQi commented on hypervisor/arch/x86/trusty.c:
```diff
@@ -456,3 +460,29 @@ bool initialize_trusty(struct vcpu *vcpu, uint64_t param)
+	 * memory after using in UOS(vSBL and/or OSloader).
+	 */
+	if (is_vm0(vm)) {
+		memcpy_s(hva, RPMB_KEY_LEN,
```
Yes, if my "refine of copy_to_vm()" be approved. Otherwise, only the first element will be copied.
